packages/firebase_messaging/firebase_messaging/example/test_driver/instance_e2e.dart

```diff
@@ -89,16 +89,17 @@ void runInstanceTests() {
       });
     }, skip: SKIP_MANUAL_TESTS); // only run for manual testing
 
-    group('deleteToken()', () {
-      test('generate a new token after deleting', () async {
-        final token1 = await messaging.getToken();
-        await messaging.deleteToken();
-        final token2 = await messaging.getToken();
-        expect(token1, isA<String>());
-        expect(token2, isA<String>());
-        expect(token1, isNot(token2));
-      }, skip: SKIP_MANUAL_TESTS); // only run for manual testing
-    });
+    // FIXME `melos run tests:e2e` with ios simulator: [E]: MissingPluginException(No implementation found for method Messaging#deleteToken on channel plugins.flutter.io/firebase_messaging)
```

Comment:
Oh my gosh, I see the problem here, quite a bad one actually, will cause issues with `getInitialMessage`, `deleteToken`, `getAPNSToken` & `setForegroundNotificationPresentationOptions`;

This line should be an `else if` https://github.com/FirebaseExtended/flutterfire/blob/master/packages/firebase_messaging/firebase_messaging/ios/Classes/FLTFirebaseMessagingPlugin.m#L117-L118 but it's not so it's running the method specific code AND also falling back to the final `else` on the next `if else` blocks which causes the not implemented: https://github.com/FirebaseExtended/flutterfire/blob/master/packages/firebase_messaging/firebase_messaging/ios/Classes/FLTFirebaseMessagingPlugin.m#L143

Sending up a PR for this shortly

Reply:
PR #4339 merged, so you should no longer need this

Reply:
Fantastic! As I respond your comments I'll edit this out before re-pushing

Reply:
dropped this commit before re-pushing